lib/xnvme_be_linux_async_liburing.c

```diff
@@ -80,18 +80,18 @@ xnvme_be_linux_liburing_init(struct xnvme_queue *q, int opts)
 	// Ring-initialization
 	//
 	if (queue->poll_sq) {
-		iou_flags |= IORING_SETUP_SQPOLL;
+		ring_params.flags |= IORING_SETUP_SQPOLL;
 	}
 	if (queue->poll_io) {
-		iou_flags |= IORING_SETUP_IOPOLL;
+		ring_params.flags |= IORING_SETUP_IOPOLL;
 	}
 
 	if (opts & XNVME_QUEUE_IOU_BIGSQE) {
-		iou_flags |= IORING_SETUP_SQE128;
-		iou_flags |= IORING_SETUP_CQE32;
+		ring_params.flags |= IORING_SETUP_SQE128;
+		ring_params.flags |= IORING_SETUP_CQE32;
 	}
 
-	err = io_uring_queue_init(queue->base.capacity, &queue->ring, iou_flags);
+	err = io_uring_queue_init_params(queue->base.capacity, &queue->ring, &ring_params);
```

Comment:
Looks good. I was comparing with FIO io_uring and it seems we are missing the new options. See these recent commits

https://github.com/axboe/fio/commit/e453f369ecf4db4913ef6916603037f2d7141035
https://github.com/axboe/fio/commit/4d22c1037dafdac3d9bf30a3fb39d88f4391e41a

especially the 2nd one, where Jens claims that it improves RR for QD8 from 680K to 870K

Reply:
I am taking a look!

Reply:
Seems like:

- ``IORING_SETUP_SINGLE_ISSUER``
- ``IORING_SETUP_DEFER_TASKRUN``

Are  liburing 2.2+, so I am creating an issue of things to address with the next liburing release.
Seems like:
- ``IORING_SETUP_COOP_TASKRUN | IORING_SETUP_TASKRUN_FLAG``
Is low-hanging fruit, man-page mentions a use-case where it is less ideal (one thread submitting another waiting on completions). But, does not say that it directly hurts performance in that case. Worth taking for a spin.

Reply:
Tried a different approach for ``IORING_SETUP_SINGLE_ISSUER``, added it under an ``ifndef`` and ``queue_init_params()``-retry in case of EINVAL. It seemed like low-hanging fruit to enable it in the context of SQPOLL done here.